sample/samplers.go

```diff
@@ -0,0 +1,87 @@
+package sample
+
+import (
+	"errors"
+	"math"
+
+	"golang.org/x/exp/rand"
```

Comment:
This package is [deprecated](https://pkg.go.dev/golang.org/x/exp/rand#pkg-overview), use `math/rand/v2` instead

Reply:
needed for weighted sample, will remove in follow up

Reply:
Let's add a TODO – ideally where `sampleuv` is used. Packages prefixed `x/exp` are for experimental packages.